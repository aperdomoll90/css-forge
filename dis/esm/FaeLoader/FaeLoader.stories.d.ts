import { FaeLoaderPropsType } from './FaeLoader.types';
declare const _default: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("@storybook/types").Args>;
export default _default;
export declare const FaeLoaderComponent: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, FaeLoaderPropsType>;
