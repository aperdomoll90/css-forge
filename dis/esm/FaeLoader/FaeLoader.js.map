{"version":3,"file":"FaeLoader.js","sourceRoot":"","sources":["../../../src/FaeLoader/FaeLoader.tsx"],"names":[],"mappings":";;;;;;;;;;;;AACA,OAAO,cAAc,CAAA;AACrB,OAAO,2BAA2B,CAAA;AAElC,MAAM,CAAC,IAAM,SAAS,GAAiC,UAAC,EAAQ;QAAN,IAAI,UAAA;IAC5D,IAAM,aAAa,GAAG;QACpB,KAAK,EAAE,UAAG,IAAI,OAAI;QAClB,MAAM,EAAE,UAAG,IAAI,OAAI;KACpB,CAAA;IACD,IAAM,SAAS,GAAG;QAChB,KAAK,EAAE,UAAG,IAAI,OAAI;QAClB,MAAM,EAAE,UAAG,IAAI,OAAI;QACnB,GAAG,EAAE,WAAI,IAAI,IAAI,IAAI,GAAG,GAAG,OAAI;KAChC,CAAA;IAED,OAAO,CACL,wBAAK,SAAS,EAAC,oBAAoB,iBACjC,cAAK,SAAS,EAAC,WAAW,EAAC,KAAK,EAAE,aAAa,GAAQ,EACvD,cAAK,SAAS,EAAC,WAAW,EAAC,KAAK,EAAE,aAAa,GAAQ,EACvD,cAAK,SAAS,EAAC,WAAW,EAAC,KAAK,EAAE,SAAS,GAAQ,EACnD,qCAAiB,KACb,CACP,CAAA;AACH,CAAC,CAAA"}