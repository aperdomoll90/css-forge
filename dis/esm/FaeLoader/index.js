export { FaeLoader } from "./FaeLoader";
export * from './FaeLoader.types';
//# sourceMappingURL=index.js.map