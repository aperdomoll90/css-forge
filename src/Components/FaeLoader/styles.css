.FaeLoaderContainer{
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;
}

.FaeLoaderContainer p{
    position: absolute;
    left: calc(50% -0.6rem);
    color: #fff;
    font-size: 1.2rem;
    text-transform: uppercase;
    letter-spacing: 0.1em;
}

.faeCircle{
    position: relative;
    margin: -30px;
    border-radius: 50%;
    animation: faeCircleRoute 4s linear infinite;
}

.faeCircle::before{
    content: "";
    width: 12px;
    height: 12px;
    border-radius: 50%;
    position: absolute;
    top:14px;
    right: 14px;
    background:#0011ff;
    box-shadow: 0 0 0 2px #0011ff33, 0 0 0 4px #0011ff22,0 0 0 8px #0011ff11;
}

@keyframes faeCircleRoute {
    0%{
        transform: rotate(0deg);
    }
    100%{
        transform: rotate(360deg);
    }
}

.faeCircle:nth-child(2){
    animation: faeCircleRoute2 4s linear infinite;
    animation-delay: -1s;
    /* border-top: 4px solid transparent;
    border-left: 4px solid #69f18644; */
}

.faeCircle:nth-child(2)::before{
    content: "";
    width: 12px;
    height: 12px;
    border-radius: 50%;
    position: absolute;
    top: initial;
    bottom:14px;
    left: 14px;
    background:#69f186;
    box-shadow: 0 0 0 2px #69f18633, 0 0 0 4px #69f18622,0 0 0 8px #69f18611;
}

.faeCircle:nth-child(3){
    position: absolute;
    animation: faeCircleRoute2 4s linear infinite;
    animation-delay: -2s;
    /* border-top: 4px solid transparent;
    border-left: 4px solid #d6292944; */
}
.faeCircle:nth-child(3)::before{
    content: "";
    width: 12px;
    height: 12px;
    border-radius: 50%;
    position: absolute;
    top: initial;
    bottom:14px;
    left: 14px;
    background:#d62929;
    box-shadow: 0 0 0 2px #d6292933, 0 0 0 4px #d6292922,0 0 0 8px #d6292911;
}

@keyframes faeCircleRoute2 {
    0%{
        transform: rotate(360deg);
    }
    100%{
        transform: rotate(0deg);
    }
}