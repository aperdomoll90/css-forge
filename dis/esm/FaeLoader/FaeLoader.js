var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import './styles.css';
import '../utils/GlobalStyles.css';
export var FaeLoader = function (_a) {
    var size = _a.size;
    var faeCircleSize = {
        width: "".concat(size, "px"),
        height: "".concat(size, "px"),
    };
    var topCircle = {
        width: "".concat(size, "px"),
        height: "".concat(size, "px"),
        top: "-".concat(size && size / 2.5, "px"),
    };
    return (_jsxs("div", __assign({ className: 'FaeLoaderContainer' }, { children: [_jsx("div", { className: 'faeCircle', style: faeCircleSize }), _jsx("div", { className: 'faeCircle', style: faeCircleSize }), _jsx("div", { className: 'faeCircle', style: topCircle }), _jsx("p", { children: "loading..." })] })));
};
//# sourceMappingURL=FaeLoader.js.map