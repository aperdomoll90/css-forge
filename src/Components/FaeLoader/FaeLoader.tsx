import './styles.css'

export interface FaeLoaderPropsType {
  size?: number;
}

function FaeLoader({size}:FaeLoaderPropsType) {

    const faeCircleSize ={
        width:`${size}px`,
        height:`${size}px`,
    }
    const topCircle ={
        width:`${size}px`,
        height:`${size}px`,
        top:`-${size && size/2.5}px`,
    }
  
  return (
    <div className='FaeLoaderContainer'>
      <div className="faeCircle" style={faeCircleSize}></div>
      <div className="faeCircle" style={faeCircleSize}></div>
      <div className="faeCircle" style={topCircle}></div>
      <p>loading...</p>
    </div>
  )
}

export default FaeLoader


