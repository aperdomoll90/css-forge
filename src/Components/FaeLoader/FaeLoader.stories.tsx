import { Meta, StoryFn } from '@storybook/react'
import FaeLoader, { FaeLoaderPropsType } from './FaeLoader'
import StorybookContainer from '../StorybookContainer'

export default {
  title: 'Loaders',
  component: FaeLoader,
  parameters: {
    jest: ['FaeLoader.test.tsx'],
  },
  argTypes: {
    size: { options: [90, 80, 50,], control: { type: 'radio' } },
  },
} as Meta

const FaeLoaderTemplate: StoryFn<FaeLoaderPropsType> = (args) => (
  <StorybookContainer>
    <FaeLoader {...args} />
  </StorybookContainer>
)

export const FaeLoaderComponent = FaeLoaderTemplate.bind({})
FaeLoaderComponent.args = {
  size: 80,
}