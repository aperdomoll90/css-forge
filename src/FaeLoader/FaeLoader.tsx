import { FaeLoaderPropsType } from './FaeLoader.types'
import './styles.css'
import '../utils/GlobalStyles.css'

export const FaeLoader: React.FC<FaeLoaderPropsType> = ({ size }) => {
  const faeCircleSize = {
    width: `${size}px`,
    height: `${size}px`,
  }
  const topCircle = {
    width: `${size}px`,
    height: `${size}px`,
    top: `-${size && size / 2.5}px`,
  }

  return (
    <div className='FaeLoaderContainer'>
      <div className='faeCircle' style={faeCircleSize}></div>
      <div className='faeCircle' style={faeCircleSize}></div>
      <div className='faeCircle' style={topCircle}></div>
      <p>loading...</p>
    </div>
  )
}
