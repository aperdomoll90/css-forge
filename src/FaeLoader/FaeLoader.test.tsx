import { render } from '@testing-library/react'
import {FaeLoader} from './FaeLoader'
import { FaeLoaderPropsType } from './FaeLoader.types'


describe('FaeLoader component', () => {
    // const { axe, toHaveNoViolations } = require('jest-axe')
    // expect.extend(toHaveNoViolations)

    let props:FaeLoaderPropsType = {
      size: 10,
}
  const renderComponent = () => render(<FaeLoader {...props} />)
  it('should match snapshots', () => {

    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
//   it('should demonstrate ADA compliance on this component', async () => {
//     const { container } = renderComponent()
//     const results = await axe(container)
//     expect(container).toHaveNoViolations()
//   })
})
