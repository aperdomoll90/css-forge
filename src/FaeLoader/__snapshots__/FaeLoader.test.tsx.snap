// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaeLoader component should match snapshots 1`] = `
<div>
  <div
    class="FaeLoaderContainer"
  >
    <div
      class="faeCircle"
      style="width: 10px; height: 10px;"
    />
    <div
      class="faeCircle"
      style="width: 10px; height: 10px;"
    />
    <div
      class="faeCircle"
      style="width: 10px; height: 10px; top: -4px;"
    />
    <p>
      loading...
    </p>
  </div>
</div>
`;
